src/main/java/space/obminyashka/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -64,11 +69,21 @@ public Page<AdvertisementTitleDto> findThumbnails(AdvertisementFindRequest findA
                         PageRequest.of(findAdvsRequest.getPage(), findAdvsRequest.getSize()))
                 .map(advertisementMapper::toAdvertisementTitleDto);
     }
+
     @Override
     public Page<AdvertisementTitleDto> findAllFavorite(String username, Pageable pageable) {
         return advertisementRepository.findFavoriteAdvertisementsByUsername(username, pageable)
                 .map(advertisementMapper::toAdvertisementTitleDto);
     }
+
+    @Override
+    public void deleteFavorite(UUID advertisementId, String username) {
+        if (advertisementRepository.deleteFavoriteAdvertisementByAdvIdAndUsername(advertisementId, username) == 0) {
```

Comment:
I'd recommend you to extract the whole construction into a new variable that would better describe what's you comparing with